### PR TITLE
HTTP Put, Patch, Delete

### PR DESCRIPTION
**Pull request by Squab**

### CODE REVIEW

File: tests/test_spag.py [hunk 1]
```diff
@@ -120,3 +129,51 @@ def test_spag_post(self):
...
+                                 '-H', 'content-type: application/json')
+        self.assertEquals(ret, 0)
+        self.assertEquals(json.loads(out), {"id": "b"})
+        self.assertEquals(err, '')
+
+class TestPatch(BaseTest):
+
+    def test_spag_patch(self):
+        run_spag('set', ENDPOINT)
+        out, err, ret = run_spag('post', '/things', '--data', '{"id": "a"}',
```
Comment (TimSimmons) [inline tests/test_spag.py]: Don't particularly like having to POST just to test PATCH...is there a better way?

Comment (pglass) [inline tests/test_spag.py]: You could change the test app to do whatever you want on a PATCH, but it's not a big deal.

**Comment by pglass** [comment]
Needs rebase?

**Comment by pglass** [comment]
Didn't need rebase.

**Pull request merged**